frb/surveys/catalog_utils.py

```diff
@@ -165,6 +166,114 @@ def summarize_catalog(frbc, catalog, summary_radius, photom_column, magnitude):
     # Return
     return summary_list
 
+def _detect_mag_cols(photometry_table):
+    """
+    Searches the column names of a 
+    photometry table for columns with mags.
+    Args:
+        photometry_table: astropy Table
+            A table containing photometric
+            data from a catlog.
+    Returns:
+        mag_colnames: list
+            A list of column names with magnitudes
+        mag_err_colnames: list
+            A list of column names with errors
+            in the magnitudes.
+    """
+    assert type(photometry_table)==Table, "Photometry table must be an astropy Table instance."
+    allcols = np.array(photometry_table.colnames)
+
+    #Start with DES
+    photom_cols = []
+    photom_errcols = []
+
+    descols = [col for col in allcols if "DES_" in col and "tile" not in col and "ID" not in col and "err" not in col]
+    des_errcols = [col for col in allcols if "DES_" in col and "err" in col]
+    
+    photom_cols += descols
+    photom_errcols += des_errcols
+
+    #WISE
+    wisecols = [col for col in allcols if ("WISE" in col or "W" in col) and "err" not in col]
+    wise_errcols = [col for col in allcols if ("WISE" in col or "W" in col) and "err" in col]
+
+    photom_cols += wisecols
+    photom_errcols += wise_errcols
+
+    #SDSS
+    sdsscols = [col for col in allcols if "SDSS" in col and "err" not in col and "field" not in col and "ID" not in col]
+    sdss_errcols = [col for col in allcols if "SDSS" in col and "err" in col]
+
+    photom_cols += sdsscols
+    photom_errcols += sdss_errcols
+
+    #Pan-STARRS
+    pscols = [col for col in allcols if "Pan-STARRS" in col and "err" not in col and "ID" not in col]
+    ps_errcols = [col for col in allcols if "Pan-STARRS" in col and "err" in col]
+
+    photom_cols += pscols
+    photom_errcols += ps_errcols
+
+    #DECaLS
+    decalcols = [col for col in allcols if "DECaL" in col and "err" not in col and "ID" not in col and 'brick' not in col and 'W' not in col]
+    decal_errcols = [col for col in allcols if "DECaL" in col and "err" in col and 'W' not in col]
+
+    photom_cols += decalcols
+    photom_errcols += decal_errcols
+
+    return photom_cols,photom_errcols
+
+def convert_mags_to_flux(photometry_table,fluxunits=units.mJy):
+    """
+    Takes a table of photometric measurements
+    in mags and converts it to flux units.
```

Comment:
Is it assuming all mags are in the AB frame?
Worth noting.

Reply:
Yes but with the exception of WISE mags. I'm using the specific zero points for each of the bands in WISE.